File: https:/github.com/kuldeeppatel-cg/Leetcode.git/169-MajorityElement/169-MajorityElement.js
// Last updated: 17/07/2026, 14:20:05
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let obj = {};
   for(let i=0; i<nums.length;i++){
      if(obj[nums[i]] != null){
         obj[nums[i]]++;
      }
      else{
         obj[nums[i]] = 1;
      }
   }
   let max = 0;
   for(val in obj){
     if(obj[val] > max){
       max = obj[val];
       var res = parseInt(val);
     }
   }
   return res;
};

File: https:/github.com/kuldeeppatel-cg/Leetcode.git/169-MajorityElement/169-MajorityElement.cpp
// Last updated: 17/07/2026, 14:20:01
class Solution {
public:
    int majorityElement(vector<int>& nums) {
     unordered_map<int,int> mp;
     for(int value: nums) {
        mp[value]++;
     }
     int freq = nums.size()/2;

     for(auto& it: mp){
        if(it.second > freq){
            return it.first;
        }
     }
     return -1;
}
};
